docs/sources/admin/troubleshooting.md

```diff
@@ -1 +1,86 @@
+
+```sql
+SELECT role.name
+FROM "user"
+JOIN "roles_users" ON ("user".id = "roles_users".user_id)
+JOIN "role" ON ("role".id = "roles_users".role_id)
+WHERE email = 'thcrockett@uchicago.edu';
+```
+
+```
+name            
+----------------------------
+default_hmis_service_stays
+default_jail_bookings
+(2 rows)
+```
+
+In this example, the user `thcrockett@uchicago.edu` has access to the `default_hmis_service_stays` and `default_jail_bookings` roles, which in matching tool parlance translates to `hmis_service_stays` and `jail_bookings` event types for the jurisdiction named `default`. This user should be seeing buttons on the upload page for both of those event types.
+
+If the user does not have entries here, you may add them, either through the [create user script](updating.md) or through directly modifying the database.
+
+If the user *does* have entries here, refer to the [docker logs](troubleshooting.md#docker-logs) in the webapp container around when they were using the application. A possible cause of this problem is if the event types they are a part of don't match the schemas listed by the app. For the most up to date list of event types, refer to `webapp/schemas/uploader/`. The filenames in that directory (dashes replaced with underscores) are queried by the app and make up a whitelist of event types. Any event type the user is associated with that doesn't match that list will be ignored by the application.
+
+## Upload Issues
+
+### System failure
+
+Although the majority of messages that users will see coming from the upload portion of the tool should be *validation* errors (user-fixable, referenced in the [Users](/users/using.md) section of the guide), occasionally they may see a message that indicates a 'System Failure'/'System Error' and they need to come to you for help. In this case, there is something that failed during the upload process in a way that the system did not expect. You should look at the [docker logs](troubleshooting.md#docker-logs) for both the webapp and `webapp_worker` containers that cover the period that the user was using the tool. If there's a lot of output, you can filter to 'ERROR' messages (e.g. `docker logs webapp | grep ERROR`), though often the messages surrounding an error message will give helpful context.
+
+
+## Matching Issues
+
+### Matching Failure
+
+If a user reports that the app's timeline indicates that a matching job failed, you can look at the [docker logs](troubleshooting.md#docker-logs) for the `matcher_worker` container. Warning: The matcher outputs a **lot** of log messages, you will likely want to heavily filter the logs. For instance, just looking at the last 100 lines or so (e.g. ``docker logs matcher_worker --tail 100`` would let you see any errors that were thrown and stopped the matching job.
```

Comment:
One thing that could happen is the configuration file could be wrong. So, if the matcher tries to make features on a column that isn't in the file, it will crash. I'll need to look up what error this generates.